standard_sequence/classy_sequence.py

```diff
@@ -765,38 +869,149 @@ def _tweezer_modulation_sequence(self, t):
-            bias_field_vector=(
-                shot_globals.mw_biasx_field,
-                shot_globals.mw_biasy_field,
-                shot_globals.mw_biasz_field,
-            ),
-        )
+        t += 3e-3
 
-        t += self.BField_obj.CONST_COIL_OFF_TIME
+        if shot_globals.do_depump_pulse_before_pumping:
+            t = self.depump_ta_pulse(t)
 
-        # ramp down power before microwave
-        t = self.TweezerLaser_obj.ramp_power(t, shot_globals.tw_ramp_dur, shot_globals.tw_ramp_power)
-        if shot_globals.do_mw_pulse:
-            t = self.Microwave_obj.do_pulse(t, shot_globals.mw_time)
-                # ramp up to full power before imaging
+        if shot_globals.do_op:
+            t, t_aom_off = self.pump_to_F4(
+                t, shot_globals.op_label, close_all_shutters=True
+            )
+        elif shot_globals.do_dp:
+            t, t_aom_off = self.depump_to_F3(
+                t, shot_globals.op_label, close_all_shutters=False
+            )
 
+        if shot_globals.op_label == "mot":
+            if shot_globals.do_depump_pulse_after_pumping:
+                t_aom_off = self.depump_ta_pulse(t_aom_off)
 
-        if shot_globals.do_killing_pulse:
-            #Labscript screws up the ramp if close_all_shutters is false...
-            t = self.kill_F4(t, close_all_shutters = True)
+            mw_biasx_field, mw_biasy_field, mw_biasz_field = (
+                self.BField_obj.convert_bias_fields_sph_to_cart(
+                    shot_globals.mw_bias_amp,
+                    shot_globals.mw_bias_phi,
+                    shot_globals.mw_bias_theta,
+                )
+            )
+
+            # [mw_biasx_field, mw_biasy_field, mw_biasz_field] = [
+                #     shot_globals.mw_biasx_field,
+                #     shot_globals.mw_biasy_field,
+                #     shot_globals.mw_biasz_field,
+                # ]
+
+            t = self.BField_obj.ramp_bias_field(
+                    t_aom_off,
+                    bias_field_vector=(mw_biasx_field, mw_biasy_field, mw_biasz_field),
+                    dur=shot_globals.mw_bias_ramp_dur,
+                )
 
-        t = self.TweezerLaser_obj.ramp_power(t, shot_globals.tw_ramp_dur, 1-1e-10)
-        t += 1e-3 # TODO: from the photodetector, the optical pumping beam shutter seems to be closing slower than others
-            # that's why we add extra time here before imaging to prevent light leakage from optical pump beam
 
+            # Making sure there is enough time between the MOT pumping pulse and the killing pulse to switch shutters
+            t = t + max(
+                D2Lasers.CONST_MIN_SHUTTER_ON_TIME
+                + D2Lasers.CONST_SHUTTER_TURN_ON_TIME
+                - shot_globals.tw_ramp_dur,
+                0,
+            )
+            t += shot_globals.mw_field_wait_dur
+            t = self.TweezerLaser_obj.ramp_power(
+                t, shot_globals.tw_ramp_dur, shot_globals.tw_ramp_power
```

Comment:
This is trying to make sure when the ramp of tweezer end (it reaches the minimum power), the shutter config is already switched to optical pumping if the tweezer ramp is too quick, it will wait extra time at high power before the shutter switching finishes. Comment added.